sql/privilege/privilege.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package privilege
+
+import (
+	"sort"
+	"strings"
+)
+
+// Kind defines a privilege. This is output by the parser,
+// and used to generate the privilege bitfields in the PrivilegeDescriptor.
+type Kind uint32
+
+// List of privileges. ALL is specifically encoded so that it will automatically
+// pick up new privileges.
+// TODO(marc): deprecate READ|WRITE.
+const (
+	_        = iota
+	ALL Kind = iota
+	CREATE
+	DROP
+	GRANT
+	SELECT
+	INSERT
+	DELETE
+	UPDATE
+	READ
+	WRITE
+)
+
+// nameMap maps privilege kinds to their names.
+var nameMap = map[Kind]string{
```

Comment:
done. interestingly, go generate fails if I have any calls to Kind.String(). I had to comment them out, generate. Then un-comment and generate again works.